src/main/java/attendance/controller/AttendanceController.java

```diff
@@ -1,122 +1,119 @@
+        option.put(RECORD, this::recordAttendance);
+        option.put(EDIT, this::editAttendance);
+        option.put(CHECK_RECORD, this::checkRecords);
+        option.put(CHECK_PENALTY, this::checkPenalty);
     }
 
-    private void validateAttendanceDate(LocalDate attendDate) {
-        if (DateUtil.isWeekend(attendDate) || Holiday.isHoliday(attendDate)) {
-            throw new IllegalArgumentException(String.format("%n[ERROR] %s은 등교일이 아닙니다.", attendDate.format(
-                DateTimeFormatter.ofPattern(OutputView.DATE_FORMATTER, Locale.KOREAN))));
+    private void execute(String inputFunction) {
+        Runnable runnable = option.getOrDefault(inputFunction, null);
+        if (runnable == null) {
+            throw new IllegalArgumentException(ErrorMessage.INVALID_INPUT_OPTION_ERROR.getMessage());
         }
+        runnable.run();
     }
 
     private void recordAttendance() {
-        Crew crew = getCrew();
-        LocalTime checkInTime = getCheckInTime();
-        LocalDateTime attendanceDateTime = LocalDateTime.of(LocalDate.now(), checkInTime);
-        Attendance attendance = Attendance.of(attendanceDateTime);
-        attendancesBook.addAttendance(crew, attendance);
-        OutputView.printAttendanceResult(attendance);
-    }
+        validateDate(systemDate);
+        Crew crew = getCrew(InputView.readNickname());
 
-    private Crew getCrew() {
-        String inputNickName = InputView.readNickName();
-        return crews.getCrew(inputNickName);
-    }
+        LocalTime inputAttendTime = InputView.readAttendTimeForRecord();
+        Attendance attendance = new Attendance(systemDate, inputAttendTime);
 
-    private LocalTime getCheckInTime() {
-        String inputCheckInTime = InputView.readCheckInTime();
-        return LocalTime.parse(inputCheckInTime);
+        attendanceBook.add(crew, attendance);
+        OutputView.printRecordAttendanceResult(attendance);
     }
 
-    private void modifyAttendance() {
-        String nickName = InputView.readModifyingNickName();
-        Crew crew = crews.getCrew(nickName);
-        LocalDate modifyingCheckInDate = getModifyingCheckInDate();
-        validateAttendanceDate(modifyingCheckInDate);
-        LocalTime modifyingCheckInTime = getModifyingCheckInTime();
+    private void validateDate(LocalDate inputDate) {
+        if (DateUtil.isWeekend(inputDate) || Holiday.isHoliday(inputDate)) {
+            throw new IllegalArgumentException(FormattedErrorMessage.INVALID_ATTEND_DATE_ERROR.getDateFormatMessage(systemDate));
+        }
+    }
 
-        Attendance previousAttendance = attendancesBook.getExistAttendanceOfCrew(crew, modifyingCheckInDate);
-        Attendance modifiedAttendance = attendancesBook.modify(crew, previousAttendance, modifyingCheckInTime);
-        OutputView.printModifyingResult(previousAttendance, modifiedAttendance);
+    private Crew getCrew(String inputNickname) {
+        Crew crew = new Crew(new Nickname(inputNickname));
```

Comment:
```suggestion
        Nickname nickname = new Nickname(inputNickname);
        nickname.validateBlankOrNull(nickname);

        Crew crew = new Crew(nickname);
```
저는 이런 식의 코드를 선호하는 편이에요.
객체를 생성하면서 내부에서 validation이 들어갈 것이라고 기대하지 않아서인 것 같아요.

Reply:
```java
Nickname nickname = new Nickname(inputNickname);
Crew crew = new Crew(nickname);
```
위와 같이 닉네임 생성을 명시하고 크루에 주입해주는 것에는 동의합니다.
다만 ``validateBlankOrNull`` 을 생성 시에 검증하지 않고 public으로 열게 되면 생성하는 곳에서 매 번 validation을 호출해주어야 하는걸까요?

Reply:
저는 그걸 선호하는데요~! 모든 생성에서 유효성 검증이 필요하다면 생성자 내부에 넣을 수도 있을 것 같아요. 😄